index.js

```diff
@@ -195,20 +197,21 @@ Patcher.prototype.readPatchFiles = function readPatchFiles(callback) {
 
   this.patches = {}
 
-  fs.readdir(this.options.dir, (function(err, files) {
+  var globSpec = path.join(this.options.dir, this.options.filePrefix) + '-*-*.sql'
+  glob(globSpec, (function(err, files) {
     if (err) return callback(err)
 
-    files = files.map((function(filename) {
-      return path.join(this.options.dir, filename)
-    }).bind(this))
-
     async.eachLimit(
       files,
       10,
       (function(filename, done) {
+
+        // We're still doing a check here, since the glob above could get filenames like 'patch-this-001-002.sql' where
+        // the prefix was set as 'patch or 'patch-foo-bar.sql' where the globs are not numeric.
         var m = filename.match(/-(\d+)-(\d+)\.sql$/)
```

Comment:
Yeah, I should probably check that out. :) Will add a couple more tests and which should satisfy your other comment below. Ta.